testplan/web_ui/testing/src/AssertionPane/AssertionTypes/basicAssertionUtils.js

```diff
@@ -541,35 +543,126 @@ function prepareDictCheckContent(assertion, defaultContent) {
  */
 function prepareAttachmentContent(assertion, defaultContent) {
   const paths = window.location.pathname.split('/');
-  let downloadLink;
-
-  // When running the development server, the real Testplan back-end is not
-  // running so we can't GET the attachment. Stick in a button that
-  // gives a debug message instead of the real link.
-  if ((paths.length >= 2) && (paths[1] === '_dev')) {
-    downloadLink = (
-      <button onClick={() => alert("Would download: " + assertion.dst_path)}>
-        {assertion.orig_filename}
-      </button>
-    );
-  } else if (paths.length >= 3) {
-    const uid = paths[2];
-    downloadLink = (
-      <a href={`/api/v1/reports/${uid}/attachments/${assertion.dst_path}`}>
-        {assertion.orig_filename}
-      </a>
-    );
+  const uid = paths[2];
+  let attachment_content;
+  let file_type = assertion.orig_filename.split('.').pop()
+  let file_path = assertion.dst_path;
+  let description = assertion.description;
+  let get_path = `/api/v1/reports/${uid}/attachments/${file_path}`
+
+  switch(file_type){
+    case 'txt':
```

Comment:
Nice use of switch statement. Could potentially add some more text file extensions here - .log, .out and .err are reasonably common.